src/NuGet.Jobs.Catalog2Registration/Schema/EntityBuilder.cs

```diff
@@ -169,6 +169,15 @@ public void UpdateLeafItem(RegistrationLeafItem leafItem, HiveType hive, string
 
             catalogEntry.Title = packageDetails.Title ?? string.Empty;
             catalogEntry.Version = parsedVersion.ToFullString();
+
+            if (packageDetails.Vulnerabilities != null && packageDetails.Vulnerabilities.Count > 0)
+            {
+                catalogEntry.Vulnerabilities = packageDetails.Vulnerabilities;
```

Comment:
The spec says:

> we will be inserting vulnerability info into the gz-semver2 hive only

It looks like this inserts the vulnerabilities information to all hives currently

Reply:
Nice catch again! Thanks, @loic-sharma.